src/eval/fixpoint.rs

```diff
@@ -42,8 +47,16 @@ pub fn patch_field(
         let mut thunk = env
             .get(var_id)
             .ok_or_else(|| EvalError::UnboundIdentifier(var_id.clone(), rt.pos))?;
-        thunk.borrow_mut().env.extend(rec_env.iter().cloned());
+
+        let extension: Vec<_> = rec_env
+            .iter()
+            .filter(|(id, _)| thunk.deps().map(|fv| fv.contains(id)).unwrap_or(true))
+            .cloned()
+            .collect();
```

Comment:
This vec allocation frustrates me so much...
Here, since deps is in an Rc, I think I would make a `deps_clone` in `Thunk` methods to get it directly without the `Ref`, so that I could pass the iterator directly to the extend.

Reply:
That's funny because in the meantime (last commits) I added a comment before your review to explain it. The problem here is that `deps` is borrowing the thunk, so we can't keep a lazy iterator and use it in the `extend`, because the `borrow_mut()` would fail saying there is already an immutable borrow active on the thunk's `refcell`.

Reply:
Sorry, I haven't read your comment carefully. But then your are cloning the `deps` while this code doesn't. Arguably the `deps` should be smaller than the `rec_env`, but I don't think there is a case between the two that is obviously better in every respect between the two. Also, I'm not sure to see how that would go: even with a `deps_clone`, you still have to call it on `thunk`, which borrows thunk, which would still fail when doing it while there's the mutable borrow active? What would be the difference with just calling `thunk.deps().clone()`?

Reply:
You can ignore my previous comments, I was mistaken. `deps` should way smaller, and its size is bounded by the one of `rec_env`, so you're right, it's better to clone `deps()` than the full `rec_env`.

Reply:
I'm not sure I was understandable, because in my explanation, I wanted to write the the deps because it is in a Rc, not the reference to deps that you get here.
Still not sure I'm understandable...

Reply:
Nope, but that works too :D

Reply:
Ok, now I get it. We can use an `Rc` indeed and not clone anything. I don't expect it to make a big difference, but it does make the code simpler by getting rid of the `D` parameter of `ThunkDeps` and the `cloned()` method, so here it is: f8d24a2637e8ba445a2c35f0b978af2abc5805e4